extensions/guacamole-auth-cas/src/main/java/org/apache/guacamole/auth/cas/AuthenticationProviderService.java

```diff
@@ -81,11 +86,13 @@ public CASAuthenticatedUser authenticateUser(Credentials credentials)
         if (request != null) {
             String ticket = request.getParameter(CASTicketField.PARAMETER_NAME);
             if (ticket != null) {
-                Map<String, String> tokens = ticketService.validateTicket(ticket, credentials);
+                TokensAndGroups tokensAndGroups = ticketService.validateTicket(ticket, credentials);
```

Comment:
I'm not a fan of defining a new class just for the sake of returning a pair of values when there's no higher level concept behind that class besides being a pair. `TokensAndGroups` is a type-specific pair, but that's about it. There has to be a better way.

Perhaps `validateTicket()` should return the `CASAuthenticatedUser` instance?

Reply:
Nick had suggested that I pass the tokens up and deal with groups a level up.  I felt that makes one unnecessarily dependent on the other (more fragile) and less obvious as to function.  It also ensures that we iterate over these items multiple times (less efficient).  When I get the time to do so, I'll look at and propose alternatives to this discussion.  Classes appear to be very efficient both  performance-wise, and to show what's going on.  I'm not sure why both you and Nick are not fans of using them, but I'm willing to try to work around that.

Reply:
I think my record will show that I like classes as much as the next guy, as I've implemented plenty of them! ;-)

In this case, though, it just seems like a lot of overhead to essentially accomplish splitting out the contents of an array into groups and other stuff.  To me it is simpler and more maintainable to have a private method within the existing class accomplish the same thing.

Just my two bits...okay, I've put in way more than that on this topic...